codec-http2/src/main/java/io/netty/handler/codec/http2/HpackHuffmanDecoder.java

```diff
@@ -231,8 +231,15 @@ AsciiString end() throws Http2Exception {
 
         private void append(int i) {
             if (bytes.length == index) {
-                // Always just expand by INITIAL_SIZE
-                byte[] newBytes = new byte[bytes.length + initialCapacity];
+                final int newLength;
+                // Choose an expanding strategy depending on how big the buffer already is.
+                if (bytes.length >= 1024) {
```

Comment:
Consider leaving a comment as to how this value was derived (educated guess, average header length from source X, etc..)